### PR TITLE
Add VERSION_INFO

### LINKED ISSUE #39795: add tf.version.version_info similar to sys.version_info
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: 
*NA*
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 
*CentOS 7*
- **TensorFlow installed from (source or binary)**:
*binary: pip*
- **TensorFlow version (use command below)**:
*2.3.0-dev20200522*
- **Python version**:
*3.7.6*

- **Exact command to reproduce**:
*Feature request*

### Describe the problem
It is inconvenient to parse `tf.version.VERSION` to do version-specific logic.  The request is to provide a function more like `sys.version_info` so that the consumer doesn't need to parse a string.

### Source code / logs
Consider the output of `tf.version.VERSION`:
```
tf.version.VERSION
'2.3.0-dev20200522'
```
Contrast that output with `sys.version_info`:
```
>>> import sys
>>> sys.version_info
sys.version_info(major=3, minor=7, micro=6, releaselevel='final', serial=0)
```

The request is to add a function that will return a object similar to `sys.version_info`'s output.  Parsing that string is harder than first imagined.  Notice that not only are the "fields" concatenated together, depending on the build, the fields are different.  For instance, an older release gives `2.1.0`.

to be consistent with the other names in tf.version, I propose `tf.version.VERSION_INFO`.  What fields? major, minor, and micro are a must.  However since it is dictionary-like, it wouldn't hurt to have all versions (such as git and compiler) also included.

No workaround is requested.

### PR DESCRIPTION
**Pull request by bhack**
Fixes https://github.com/tensorflow/tensorflow/issues/39795

/cc @mihaimaruseac

### CODE REVIEW

File: tensorflow/python/framework/versions.py [hunk 1]
```diff
@@ -34,6 +35,12 @@
     "__version__",
     v1=["version.VERSION", "VERSION", "__version__"]).export_constant(
         __name__, "VERSION")
+VERSION_INFO = type('VersionInfo', (object,),__version_info__)
+tf_export(
+    "version.VERSION_INFO",
+    "__version_info__",
+    v1=["version.VERSION_INFO", "VERSION_INFO", "__version_info__"])(
+        VERSION_INFO)
```
Comment (mihaimaruseac) [inline tensorflow/python/framework/versions.py:43]:
```suggestion
    "version.VERSION_INFO",
    "__version_info__")(
        VERSION_INFO)
```

Comment (mihaimaruseac) [reply tensorflow/python/framework/versions.py:43]: We don't need to export separately for v1 (in fact, v1 bindings will be removed at some point in the future if they are the same as the v2)

**Review by mihaimaruseac — APPROVED** [review]
Minor nit but LGTM

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/python/framework/versions.py (hunk 1 above)
Comment (bhack) [reply tensorflow/python/framework/versions.py]: Yes but it was just to be consistent in style  (see `VERSION` etc..)

**Comment by bhack** [comment]
I've added the golden update for v2.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by bhack** [comment]
Do we need to change that PR label here?

**Comment by mihaimaruseac** [comment]
It's pending API onwers review, hopefully we'll get it today.

**Comment by mihaimaruseac** [comment]
So API owners suggested to use `semver` or `distutils.version.LooseVersion` instead of adding a new API.

Means the workaround in https://github.com/tensorflow/tensorflow/issues/39795#issuecomment-751472512 is the recommended way.

Sorry that it gets to be closed after such a long wait.

**Comment by bhack** [comment]
@mihaimaruseac As the gist was mine as this PR do you will accept a semver dep PR?

**Comment by mihaimaruseac** [comment]
I think we should ask the API owners on the new PR. I'll add tags to have them notified

**Comment by bhack** [comment]
> I think we should ask the API owners on the new PR. I'll add tags to have them notified

Thanks I suppose that they meant to push this on the user side (outside Tensorflow) but just to be sure that we don't want a python only PR

**Comment by mihaimaruseac** [comment]
Yes, I think the same, that they wanted to have users that needed this implement on their code instead of relying on TF.

**Pull request closed without merging**